### PR TITLE
feat(tracer): Add shutdown hook for tracer

### PR DESCRIPTION
**Pull request by dougEfresh**
See #935 
Tracer is not shutdown gracefully and thus traces are lost. 
I have added tracer to the FX lifecycle and created a `WithTracing` function in `nodebuilder/settings.go`

### CODE REVIEW

File: cmd/celestia/main.go [hunk 1]
```diff
@@ -29,8 +31,18 @@ func init() {
...
 }
 
+func shutdownTracer() {
+	tp, ok := otel.GetTracerProvider().(*tracesdk.TracerProvider)
+	if ok {
+		// Flush and shutdown traces to capture all spans.
+		_ = tp.ForceFlush(context.Background())
+		_ = tp.Shutdown(context.Background())
+	}
+}
```
Comment (Wondertan) [inline cmd/celestia/main.go:41]: This logic ideally should live next to the logic configuring the tracer, `cmd/flags_misc.go`. Alternatively, we can move this to FX, as you've mentioned, but we will also need to move the _setup_ logic to the FX as we do for metrics(`nodebuilder/settings.go`)

Comment (Wondertan) [reply cmd/celestia/main.go:41]: We can consider similar `WithTraces` option

Comment (dougEfresh) [reply cmd/celestia/main.go]: @Wondertan see latest change using FX lifecycle.

**Comment by Wondertan** [comment]
Needs rebase

**Comment by renaynay** [comment]
@dougEfresh do you want to rebase this or should we take over the PR?

**Comment by dougEfresh** [comment]
> @dougEfresh do you want to rebase this or should we take over the PR?

Will rebase soon

**Comment by ramin** [comment]
hey @dougEfresh do you want to bring this one home? I can take over if not

File: cmd/start.go [hunk 2]
```diff
@@ -60,6 +60,8 @@ Options passed on start override configuration options only on start and are not
 			defer cancel()
 			err = nd.Start(ctx)
 			if err != nil {
+				stopErr := nd.Stop(ctx)
```
Comment (Wondertan) [inline cmd/start.go:63]: This is out of the scope of this PR and unlikely to give us anything, so lets remove this

**Review by Wondertan — CHANGES_REQUESTED** [review]

**Review by Wondertan — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**